block-producer/src/block_builder/mod.rs

```diff
@@ -82,16 +83,17 @@ where
 
         let updated_accounts: Vec<_> =
             batches.iter().flat_map(TransactionBatch::updated_accounts).collect();
-        let created_notes = batches
+
+        let created_notes: Vec<(usize, usize, NoteCreated)> = batches
```

Comment:
We don't need these two additional `usize`s, because we already have `batch_index` and `note_index` fields in `NoteCreated` structure. The task is to fill them here. 

I think, the best solution would be to not to create `NoteCreated` in the batch builder (see comments above), but just to add serialized details field in the `TransactionBatch::created_note_envelopes` vector:
```rust
created_note_envelopes_with_details: Vec<(NoteEnvelope, Option<Vec<u8>>)>,
```
And after that we would be able to produce `NoteCreated` here and init its `batch_index` and `note_index` fields during creation.